### PR TITLE
Register

### PR DESCRIPTION
**Pull request by jupabass89**
## Summary
-The view was created as indicated on the registration card.
-The reactive form was created.
-The singUp method was created that collects data from the view and prints them in a Json.

## Screenshot
Captura de Pantalla 2019-06-28 a la(s) 12 08 22

## Trello Card
https://trello.com/c/Z6ZKVn9O/8-register-layout-exercise

### CODE REVIEW

File: wolox-books/src/app/app.component.html [hunk 1]
```diff
@@ -1,21 +1,6 @@
...
-    <h2><a target="_blank" rel="noopener" href="https://angular.io/tutorial">Tour of Heroes</a></h2>
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://angular.io/cli">CLI Documentation</a></h2>
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://blog.angular.io/">Angular blog</a></h2>
-  </li>
-</ul>
+
```
Comment (tammaroivan) [inline wolox-books/src/app/app.component.html]: remove this empty line

File: wolox-books/src/app/app.component.html [hunk 2]
```diff
@@ -1,21 +1,6 @@
...
+<a routerLink="login"></a>
+
+  <app-login></app-login>
```
Comment (tammaroivan) [inline wolox-books/src/app/app.component.html]: Check indentation

File: wolox-books/src/app/app.component.html [hunk 3]
```diff
@@ -1,21 +1,6 @@
...
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://angular.io/cli">CLI Documentation</a></h2>
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://blog.angular.io/">Angular blog</a></h2>
-  </li>
-</ul>
+
+<a routerLink="login"></a>
```
Comment (tammaroivan) [inline wolox-books/src/app/app.component.html]: what's this button for? You are including the app-login down there

File: wolox-books/src/app/login/login.component.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<div class="window">
+  <div class="login">
+    <div class="login-title">
+      <img class="login-img" src="../../assets/wolox_logo.svg" alt="wolox-logoi">
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.html]: Fix the alt and i think you can do just `src="assets/wolox_logo.svg"`

File: wolox-books/src/app/login/login.component.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      <h1 class="login-text-title"> Books</h1>
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.html]: Remove that space before `Books`

File: wolox-books/src/app/login/login.component.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    </div>
+    <form [formGroup]="form" (ngSubmit)="singUp(form.value)" class="login-form">
+      <p class="input-title">First name</p>
+      <input class="login-input" formControlName="first_name" type="text">
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.html]: use camel case for variables like "first_name"

File: wolox-books/src/app/login/login.component.html [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      <p class="input-title">Last name</p>
+      <input class="login-input" formControlName="last_name" type="text">
+      <p class="input-title">Email</p>
+      <input class="login-input" formControlName="email" type="text">
+      <p class="input-title">Password</p>
+      <input class="login-input" formControlName="password" type="password">
+      <div class="sing-up-div">
+        <input value="Sing up" class="login-input sing-up-button" type="submit">
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.html]: Use a button type submit

File: wolox-books/src/app/login/login.component.html [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      </div>
+    </form>
+    <div class="login-button-div">
+      <input value="Login" class="login-input login-button" type="button">
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.html]: Use a button for this

File: wolox-books/src/app/login/login.component.scss [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    align-items: center;
+    background-color: $login-grey;
+    border-top: 4px solid $blue;
+    display: flex;
+    flex-direction: column;
+    padding: 20px;
+    width: 330px;
+}
+
+.login-button {
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.scss]: use sass, you can dou

```
.login {
  //login base code

  .login-button-div {
    ...
  }
}
```

File: wolox-books/src/app/login/login.component.scss [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    background-color: $green;
+    border-radius: 10px;
+    color: $white; 
+    width: 100%;
+}
+
+.sing-up-div {
+    border-bottom: 2px solid $grey; 
+    height: 65px;
+    margin-bottom: 10px;width: 100%;
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.scss]: separate in 2 lines

File: wolox-books/src/app/login/login.component.scss [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.window {
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.scss]: use a better naming that `window`

File: wolox-books/src/app/login/login.component.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+  form: FormGroup;
+  post: any;
+
+  constructor(private fb: FormBuilder) { 
+  }
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.ts]: keep this in the same line than constructor

File: wolox-books/src/app/login/login.component.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+  singUp(post) {
+    var email = post.email;
+    var password = post.password;
+    var password_confirmation = post.password;
+    var first_name = post.first_name;
+    var last_name = post.last_name;
+    var locale = "en";
+    console.log(JSON.stringify({user:{email , password , password_confirmation, first_name, last_name, locale}}));
```
Comment (tammaroivan) [inline wolox-books/src/app/login/login.component.ts]: there is no necessary to assign all that variables before, you can do it here. And we **never** use `var`. Use `let` or `const`

Comment (jupabass89) [reply wolox-books/src/app/login/login.component.ts]: I tried to no define each single variable and do all this in a single line but it doesn't work, could you explain me how to do this please?

File: wolox-books/src/app/register/register.component.ts [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  singUp(post: any) {
+    let user = { 
+      email: post.email, 
+      password: post.password, 
+      password_confirmation: post.password, 
+      first_name: post.firstName, 
+      last_name: post.lastName,
+      locale:"en"
+    }
+    console.log(JSON.stringify({user}));
```
Comment (tammaroivan) [inline wolox-books/src/app/register/register.component.ts]: You can console log without sintrigify but that's ok

**Pull request merged**